### ISSUE TITLE
Toggling visibility shouldn't affect selection (segmentations, markers, models, etc...)

### ISSUE THREAD

**Issue by fusentasticus** [Slicer/Slicer#5290]
I believe this is a bug, not a feature request, in so far as the behavior is inconsistent with standard UI design (afaik).

In the Segmentation Editor, when toggling visibility of a segment, the selection should not be affected; in particular, it should not be changed to that of the segment whose visibility was changed. 

This leads (for me) to very frequent operations of the wrong segment and it's sometimes hard to discover that the operation was misapplied.

The problem is general to all of the 3D slicer UI it seems:  toggling visibility for markups, models, etc. affects the selection. I think the correct behavior is that it shouldn't.

## Environment
- Slicer version: Slicer-4.11
- Operating system: Windows / Linux / Mac + which version

**Comment by lassoan** [comment]
Good point. I've checked Blender and Gimp and they both allow changing visibility of objects/layers without making it selected. (Paraview does not change selection if you turn off visibility, but selects the object when its visibility is turned on.)

**Comment by lassoan** [comment]
We have a pull request under review (https://github.com/Slicer/Slicer/pull/5274) to warn the user if he wants to edit a non-visible segment.

**Comment by ottensmeyer** [comment]
Thanks Andras for the forum response - this appears to be the same issue - adding the comment and response here.

"In the Segment Editor, in 4.10.2, when one segment is selected, one can toggle visibility (change open/closed eye icon state) of another segment without simultaneously selecting the other segment. In 4.11, if I click the eye icon to toggle visibility of some segment, I lose focus on the segment on which I’m actually working, leading me to accidentally start editing the segment that I want to see but not edit.

In 4.10.2, hovering over the eye icon shows a dotted border around the visibility icon, which does not appear in 4.11.

Is this an intended behavior new behavior I’ll need to get used to? At the moment, I keep on clicking the icon to see another segment for reference, and start accidentally editing that one instead of the one I want to change."

lassoan:
"We reworked the segment list to add many new features (searching, filtering, states) and along the way management of checkboxes changed. I agree that the previous behavior was better, but it is not trivial to achieve that again in the new architecture. I would suggest to submit a feature request to the Slicer issue tracker."

**Comment by DIV-on-github** [comment]
In the **Segment Editor** module:  "_At the moment, I keep on clicking the icon to see another segment for reference, and start accidentally editing that one instead of the one I want to change._"  Me too — as per https://discourse.slicer.org/t/segment-editor-clicking-eye-to-toggle-visibility-should-not-select-segment/20237 .  

There's a slightly different (but related) effect that I am not so fond of in the **Volume Rendering** module too:  in the "Crop" settings within the _Display_ panel, clicking the `Display ROI` (eye) icon to display a ROI in the viewer automatically causes the `Enable` box to be ticked (enabling cropping according to that ROI) — despite the fact that there is a separate control provided.  
[On the other hand, clicking the `Display ROI` (eye) icon to _hide_ a ROI in the viewer does not affect the `Enable` setting.  Which is as it should be, in my view.]

—DIV

**Comment by pieper** [comment]
> There's a slightly different (but related) effect that I am not so fond of in the **Volume Rendering** module too: in the "Crop" settings within the _Display_ panel, clicking the `Display ROI` (eye) icon to display a ROI in the viewer automatically causes the `Enable` box to be ticked (enabling cropping according to that ROI) — despite the fact that there is a separate control provided.

FWIW, I find that behavior very useful - I almost always want to enable and display in one click, adjust the ROI, then make the ROI invisible.  I didn't implement the feature so I'm not defending my own decision it's just that I've always found it useful since it was first introduced.

**Comment by lassoan** [comment]
I did not implent the current behavior of ROI display show either, but I find it really convenient. When I enable cropping the first time I always need to display the ROI. Hiding the ROI must not disable cropping, because the ROI very often needs to be hidden (to declutter the view), while the cropping still should be applied. So, the current behavior is exactly what I need. Can you describe your workflow and how the displaying of the ROI automatically causes complications for you?

Showing/hiding segment without switching to it would be useful. That's what this issue is tracking.

**Comment by DIV-on-github** [comment]
> Showing/hiding segment without switching to it would be useful. That's what this issue is tracking.

I have transferred the other comments to https://github.com/Slicer/Slicer/issues/6126
